### PR TITLE
Add HDF5 GPU VFD package.

### PR DESCRIPTION
**Pull request by hyoklee**
This is a new package that allows users to install [HDF5 Nvidia GPUDirect Storage VFD](https://github.com/hpc-io/vfd-gds).

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    # Package info
+    homepage    = 'https://github.com/hpc-io/vfd-gds'
+    url = 'https://github.com/hpc-io/vfd-gds/archive/refs/tags/1.0.1.tar.gz'
+    git         = 'https://github.com/hpc-io/vfd-gds.git'
+    maintainers = ['hyoklee', 'lrknox']
+
+    # Versions
+    version('master', branch='master')
+    version('1.0.1', sha256='00e125fd149561be991f41e883824de826d8add604aebccf103a4fb82d5faac2', default=True)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py]:
```suggestion
    version('1.0.1', sha256='00e125fd149561be991f41e883824de826d8add604aebccf103a4fb82d5faac2')
```

File: var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+class Hdf5VfdGds(CMakePackage, CudaPackage):
+    """This pacakge enables GPU Direct Storage Virtual File Driver in HDF5."""
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py]:
```suggestion
    """This package enables GPU Direct Storage Virtual File Driver in HDF5."""
```

File: var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    git         = 'https://github.com/hpc-io/vfd-gds.git'
+    maintainers = ['hyoklee', 'lrknox']
+
+    # Versions
+    version('master', branch='master')
+    version('1.0.1', sha256='00e125fd149561be991f41e883824de826d8add604aebccf103a4fb82d5faac2', default=True)
+    version('1.0.0', sha256='6b16105c7c49f13fc05784ee69b78d45fb159270c78d760689f9cd21e230ddd2')
+    
+    # Dependencies
+    depends_on('cuda')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py]: A `CudaPackage` has already a conditional dependency on `cuda` and a `cuda` boolean variant, hence adding an unconditional dependency on `cuda` seems wrong. I think what might work here is:
```
conflicts('~cuda')
```
if the aim is to have cuda always enabled.

**Review by alalazo — CHANGES_REQUESTED** [review]

**Review by lrknox — APPROVED** [review]

**Comment by lrknox** [comment]
Note:  when built and tested on thetagpu, "make test" or --test=all failed because the default stage directory was on /tmp which was a tmpfs filesystem.  Setting TMPDIR to a lustre filesystem addressed the failure.

**Review by alalazo — APPROVED** [review]

**Pull request merged**